include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -667,9 +667,17 @@ struct __result_and_scratch_storage : __result_and_scratch_storage_base
         }
         else if (__supports_USM_device)
         {
-            _T __tmp;
-            __q.memcpy(&__tmp, __scratch_buf.get() + __scratch_n + _Idx, 1 * sizeof(_T)).wait();
-            return __tmp;
+            static_assert(sycl::is_device_copyable_v<_T>,
+                          "The type _T must be device copyable to use __result_and_scratch_storage.");
+            // Avoid default constructor for _T, we know that _T is device copyable and therefore a copy construction
+            // is equivalent to a bitwise copy.  We may treat *__tmp as if it has been constructed.
```

Comment:
Where is `__tmp` defined?

Reply:
Sorry, this was a stale piece of a comment from an earlier iteration.  fixed.

Reply:
Thanks, I had pushed to the wrong branch.